test/unit/spec/MfaVerify_spec.js

```diff
@@ -3052,6 +3128,35 @@ function (Okta,
               expect(test.form.hasErrors()).toBe(true);
             });
         });
+
+        itp('triggers an afterError event an error if u2f.sign fails', function () {
+          Q.stopUnhandledRejectionTracking();
+          var signStub = function (appId, nonce, registeredKeys, callback) {
+            callback({ errorCode: 2 });
+          };
+          return setupU2F({u2f: true, signStub: signStub})
+            .then(function (test) {
+              return Expect.waitForSpyCall(window.u2f.sign, test);
+            })
+            .then(function (test) {
+              window.u2f.tap();
+              return Expect.waitForFormError(test.form, test);
+            })
+            .then(function (test) {
+              expect(test.afterErrorHandler).toHaveBeenCalledTimes(1);
+              expect(test.afterErrorHandler.calls.allArgs()[0]).toEqual([
+                {
+                  error: jasmine.objectContaining({
+                    name: 'U2F_ERROR',
+                    message: 'There was an error with the U2F request. Try again or select another factor.'
+                  })
+                },
+                {
+                  controller: 'mfa-verify verify-u2f'
```

Comment:
might be for a follow-up jira... should we have just one name for the controller? I think only MFAVerifyController has this problem...

Reply:
Yeah - we have nested names for a few controllers currently. We're going to be looking at this with a major version rev of the Widget. I'll create a JIRA to track this though